doc/introduction/unsupported.rst

```diff
@@ -0,0 +1,343 @@
+        dev = qml.device('default.qubit', wires=1, shots=None)
+
+        @qml.qnode(dev, diff_method='backprop')
+        def circuit(x):
+            qml.RX(x[0], wires=0)
+            return qml.state()
+
+        def cost_fn(x):
+            out = circuit(x)
+            return np.abs(out[0])
+
+        x = np.array([0.1], requires_grad=True)
+        print(qml.grad(cost_fn)(x))
+
+>>> state_scalar_grad()
+[-0.02498958]
+
+However, changing from differentiating the scalar cost to differentiating the state
+directly will fail with an error:
+
+.. code-block:: python
+
+    def state_vector_grad():
+        dev = qml.device('default.qubit', wires=1, shots=None)
+
+        @qml.qnode(dev, diff_method='backprop')
+        def circuit(x):
+            qml.RX(x[0], wires=0)
+            return qml.state()
+
+        x = np.array([0.1], requires_grad=True)
+        print(qml.jacobian(circuit)(x))
+
+>>> state_vector_grad()
+Traceback (most recent call last):
+  ...
+  File "C:\Python38\lib\site-packages\numpy\core\fromnumeric.py", line 57, in _wrapfunc
+    return bound(*args, **kwds)
+ValueError: cannot reshape array of size 4 into shape (2,1)
+
+Using a different interface that supports complex differentiation will fix this error:
+
+.. code-block:: python
+
+    def state_vector_grad_jax():
+        dev = qml.device('default.qubit', wires=1, shots=None)
+
+        @qml.qnode(dev, interface='jax', diff_method='backprop')
+        def circuit(x):
+            qml.RX(x[0], wires=0)
+            return qml.state()
+
+        x = jnp.array([0.1], dtype=np.complex64)
+        print(jax.jacrev(circuit, holomorphic=True)(x))
+
+    def state_vector_grad_tf():
+        dev = qml.device('default.qubit', wires=1, shots=None)
+
+        @qml.qnode(dev, interface='tf', diff_method='backprop')
+        def circuit(x):
+            qml.RX(x[0], wires=0)
+            return qml.state()
+
+        x = tf.Variable([0.1], trainable=True, dtype=np.complex64)
+        with tf.GradientTape() as tape:
+            out = circuit(x)
+
+        print(tape.jacobian(out, [x]))
+
+    def state_vector_grad_torch():
+        dev = qml.device('default.qubit', wires=1, shots=None)
+
+        @qml.qnode(dev, interface='torch', diff_method='backprop')
+        def circuit(x):
+            qml.RX(x[0], wires=0)
+            return qml.state()
+
+        x = torch.tensor([0.1], requires_grad=True, dtype=torch.complex64)
+        print(F.jacobian(circuit, (x,)))
+
+>>> state_vector_grad_jax()
+[[-0.02498958+0.j        ]
+ [ 0.        -0.49937513j]]
+>>> state_vector_grad_tf()
+[<tf.Tensor: shape=(2, 1), dtype=complex64, numpy=
+array([[-0.02498958+0.j        ],
+       [-0.        +0.49937513j]], dtype=complex64)>]
+>>> state_vector_grad_torch()
+(tensor([[-0.0250+0.0000j],
+        [ 0.0000+0.4994j]]),)
+
+.. _Sample gradients:
+
+Sample gradients
+~~~~~~~~~~~~~~~~~~~~~~~
+
+In PennyLane, samples are drawn from the eigenvalues of an observable, or from the

```

Comment:
I've already opened an issue [here](https://github.com/PennyLaneAI/pennylane/issues/2556). Right now I labeled gradients of samples as "not supported". The page is accessible if the user wants to learn more.

I can remove the section discussing the silent failures, since I agree that it is a bug that no error is raised. What do you think?